src/main/java/com/example/librarymanagement/repositories/InventoryRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.example.librarymanagement.repositories;
+
+import com.example.librarymanagement.models.entities.Inventory;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface InventoryRepository extends JpaRepository<Inventory, Long> {
+    Optional<List<Inventory>> findAllByBookId(Long bookId);
```

Comment:
why do you need Optional<List> ?
if there is no items to return you will receive empty list